src/app/WriteHandler.cpp

```diff
@@ -253,32 +297,43 @@ CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeData
              (!mProcessingAttributePath.HasValue() || mProcessingAttributePath.Value() != dataAttributePath)))
         {
             err = AddStatus(dataAttributePath, StatusIB(Protocols::InteractionModel::Status::Busy));
+            continue;
         }
-        else
+
+        if (ShouldReportListWriteEnd(mProcessingAttributePath, mProcessingAttributeIsList, dataAttributePath))
         {
-            mProcessingAttributePath.SetValue(dataAttributePath);
-            MatterPreAttributeWriteCallback(dataAttributePath);
-            TLV::TLVWriter backup;
-            DataVersion version = 0;
-            mWriteResponseBuilder.Checkpoint(backup);
-            err = element.GetDataVersion(&version);
-            if (CHIP_NO_ERROR == err)
-            {
-                dataAttributePath.mDataVersion.SetValue(version);
-            }
-            else if (CHIP_END_OF_TLV == err)
-            {
-                err = CHIP_NO_ERROR;
-            }
-            SuccessOrExit(err);
-            err = WriteSingleClusterData(subjectDescriptor, dataAttributePath, dataReader, this);
-            if (err != CHIP_NO_ERROR)
-            {
-                mWriteResponseBuilder.Rollback(backup);
-                err = AddStatus(dataAttributePath, StatusIB(err));
-            }
-            MatterPostAttributeWriteCallback(dataAttributePath);
+            DeliverListWriteEnd(mProcessingAttributePath.Value(), true /* writeWasSuccessful */);
+        }
+
+        if (ShouldReportListWriteBegin(mProcessingAttributePath, dataAttributePath))
+        {
+            DeliverListWriteBegin(dataAttributePath);
```

Comment:
So we will do this even if the actual write will not be happening (because of missing ACLs, etc)?

This should be at least clearly documented, but ideally we would not deliver the notification if the write will not be happening....

Reply:
Still need to sort out what the right thing is here....

Reply:
I think receiving the event without touching the attribute is OK.
Considering the transactions in database, it is acceptable to start a transaction but rollback / commit without touching anything.

Reply:
However, this does reminds me about a few cases around handling the list entries, I have fixed them and added a few new tests. Thanks.